qurami/po2strings
# New Object Oriented version

This branch refactors the po2strings package using OO code, a modular approach, tests.

This change is necessary as pre-requirement to create new processors.

To test the version:
- Download the code
- Enter the directory
- Write a `it.po` test source file such as:

```po
msgid ""
msgstr ""
"Project-Id-Version: \n"
"Report-Msgid-Bugs-To: EMAIL@ADDRESS\n"
"POT-Creation-Date: 2016-11-15 15:27\n"
"PO-Revision-Date: 2016-10-19 10:52+0200\n"
"Last-Translator: Tester <tester@email.address>\n"
"Language-Team: Italian <http://address>\n"
"Language: it\n"
"MIME-Version: 1.0\n"
"Content-Type: text/plain; charset=UTF-8\n"
"Content-Transfer-Encoding: 8bit\n"
"Plural-Forms: nplurals=2; plural=n != 1;\n"
"X-Generator: Weblate 2.2\n"
"X-Poedit-Basepath: .\n"

#: source.file:2
msgctxt "K99b6b_A_SNEAK_PEEK_OF_THE_NEW_CONNECT_09e2c"
msgid "A sneak peek of the new Connect"
msgstr "Un'anteprima del nuovo Connect"

#: source.file:4
msgctxt "Kc4408_ACCEPT_d3350"
msgid "Accept"
msgstr "Accetta"

```
- Launch the processor via the following command:

`python po2strings.py it.po it.strings`

- Check that the `it.strings` file exists and it is a valid .strings file


Same should be done using a different output filename, for Android, such as `it.xml`, running the processor via 

`python po2strings.py it.po it.xml`.

Run tests via the `make test` command.

No external packages are required for this script to run.
